### PR TITLE
build(docker-image): use absolute path in ENTRYPOINT

### PR DESCRIPTION
**Pull request by jirikuncar**
- Makes sure that ENTRYPOINT uses absolute paths.

- Removes duplicate WORKDIR.

**Checklist**

- [ ] Tests added / updated
- [ ] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

> If indicated yes above, please describe the breaking change(s).
>
> **Remove this quote before creating the PR.**

**Additional context**

Allows to run the Docker image as GitHub Action with different workdir.

### CODE REVIEW

**Comment by XVincentX** [comment]
@jirikuncar Can you give some context about why is this change required? The `WORKDIR` is setting the current working directory so relative paths should not be a problem.

**Comment by jirikuncar** [comment]
The problem is when you use the image in CI which changes the `WORKDIR` (e.g. GitHub Action).

**Comment by XVincentX** [comment]
@jirikuncar I'm inclined to say there is something wrong with the hosting image or something. Can you show me an example so I can look into?

**Comment by jirikuncar** [comment]
Example from our repo with redacted paths:
```console
/usr/bin/docker run --name stoplightspectral5_d66781 --label 488dfb \
--workdir /github/workspace --rm \
-e pythonLocation -e CLIENT_REPO_NAME -e INPUT_ARGS -e INPUT_ENTRYPOINT -e HOME -e GITHUB_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_RUN_ID -e GITHUB_RUN_NUMBER -e GITHUB_ACTOR -e GITHUB_WORKFLOW -e GITHUB_HEAD_REF -e GITHUB_BASE_REF -e GITHUB_EVENT_NAME -e GITHUB_WORKSPACE -e GITHUB_ACTION -e GITHUB_EVENT_PATH -e RUNNER_OS -e RUNNER_TOOL_CACHE -e RUNNER_TEMP -e RUNNER_WORKSPACE -e ACTIONS_RUNTIME_URL -e ACTIONS_RUNTIME_TOKEN -e ACTIONS_CACHE_URL -e GITHUB_ACTIONS=true \
--entrypoint "node" -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/org/repo":"/github/workspace" \
stoplight/spectral:5 /usr/src/spectral/dist/cli/index.js lint /github/workspace/service.yaml -r /github/workspace/config/spectral.yml -v -s example-value-or-externalValue -s oas3-valid-content-schema-example -F warn
```

**Comment by jirikuncar** [comment]
You should really use absolute paths everywhere except in `RUN` commands which are guaranteed to run in defined `WORKDIR`.

File: Dockerfile [hunk 1]
```diff
@@ -32,6 +32,4 @@ COPY package.json /usr/src/spectral/
 COPY --from=compiler /usr/src/spectral/dist /usr/src/spectral/dist
 COPY --from=dependencies /usr/src/spectral/node_modules/ /usr/src/spectral/node_modules/
 
-WORKDIR /usr/src/spectral/
```
Comment (jirikuncar) [inline Dockerfile]: see line 27: https://github.com/stoplightio/spectral/pull/934/files#diff-3254677a7917c6c01f55212f86c57fbfR27

File: Dockerfile [hunk 2]
```diff
@@ -32,6 +32,4 @@ COPY package.json /usr/src/spectral/
...
-
-ENTRYPOINT [ "node", "dist/cli/index.js" ]
```
Comment (jirikuncar) [inline Dockerfile]: `docker run --workdir /my-new-workdir stoplightio/spectral ...` would not work

**Comment by XVincentX** [comment]
@jirikuncar I'm going to have a look into this as soon as possible. In meantime, did you try https://github.com/stoplightio/spectral-action instead?

**Comment by jirikuncar** [comment]
@XVincentX I know about it. Thanks anyway.

**Comment by philsturgeon** [comment]
I think this has been resolved with #1022, thank you for this!

**Pull request closed without merging**